### PR TITLE
fix: adapt all classes and tests to make the controller run

### PR DESCRIPTION
**Pull request by anarchuser**

### CODE REVIEW

File: src/model/object/entity/entity.h [hunk 1]
```diff
@@ -1,27 +1,19 @@
 //
-// Created by aaron on 26/09/2019.
+// Created by aaron on 07/10/2019.
 //
 
-/* Abstract base class for all objects in the world */
```
Comment (xeruf) [inline src/model/object/entity/entity.h:5]: why was this removed?

File: src/model/object/object.cpp [hunk 2]
```diff
@@ -1,5 +1,8 @@
 //
-// Created by aaron on 01/10/2019.
+// Created by aaron on 26/09/2019.
```
Comment (xeruf) [inline src/model/object/object.cpp:2]: wat? did you travel back in time???

File: src/controller/controller.cpp [hunk 3]
```diff
@@ -4,4 +4,12 @@
...
 
-Controller::Controller() = default;
+Controller::Controller(int WIDTH, int HEIGHT, int ENERGY) : world{World(WIDTH, HEIGHT, ENERGY)} {};
+
+void Controller::run() {
+    while (simulate()) iteration++;
+}
+
+bool Controller::simulate() {
+    // What has to be simulated
```
Comment (xeruf) [inline src/controller/controller.cpp]:
```suggestion
    // TODO What has to be simulated
```

File: src/model/model.h [hunk 4]
```diff
@@ -1,10 +1,12 @@
 //
-// Created by aaron on 26/09/2019.
+// Created by aaron on 07/10/2019.
 //
 
 #ifndef EVE_MODEL_H
 #define EVE_MODEL_H
 
+#include <exception>
```
Comment (xeruf) [inline src/model/model.h:8]: why?

File: src/model/model.cpp [hunk 5]
```diff
@@ -1,9 +1,13 @@
 //
-// Created by aaron on 26/09/2019.
+// Created by aaron on 07/10/2019.
 //
 
+#include <stdexcept>
```
Comment (xeruf) [inline src/model/model.cpp]: why?

File: test/model/world/world.cpp [hunk 6]
```diff
@@ -1,10 +1,13 @@
-#include "../../catch.hpp"
 #include "../../../src/model/world/world.h"
+#include <catch2/catch.hpp>
+#include <exception>
```
Comment (xeruf) [inline test/model/world/world.cpp:3]: shouldnt this be in the header?

File: src/model/model.cpp (hunk 5 above)
Comment (AtomToast) [inline src/model/model.cpp]: kek

File: test/model/world/world.cpp (hunk 6 above)
Comment (xeruf) [reply test/model/world/world.cpp:3]: oh nvm this is the test

File: src/model/object/entity/food/food.h [hunk 7]
```diff
@@ -6,12 +6,9 @@
 #define EVE_FOOD_H
 
 #include "../entity.h"
-#include "../../../geometry/shape/shape.h"
 
-#include <exception>
-
-#include <experimental/random>
 #include <iostream>
+#include <stdexcept>
```
Comment (AtomToast) [inline src/model/object/entity/food/food.h]:
```suggestion
```

**Review by AtomToast — DISMISSED** [review]

File: src/model/model.h (hunk 4 above)
Comment (anarchuser) [reply src/model/model.h:8]: move from source to header file

File: src/model/object/entity/entity.h (hunk 1 above)
Comment (anarchuser) [reply src/model/object/entity/entity.h:5]: this should be resolved (89239037ffa3599858f2ceb4eebac4cebaf243fa)

Comment (anarchuser) [reply src/model/object/entity/entity.h:5]: except the head containing the date, because we're gonna remove that anyways, right?

File: src/model/object/entity/food/food.h (hunk 7 above)
Comment (anarchuser) [reply src/model/object/entity/food/food.h]: resolved in ea2d8921853e0ae384be30ce03ddd5ebc85a391f

File: src/model/object/object.cpp (hunk 2 above)
Comment (anarchuser) [reply src/model/object/object.cpp:2]: just ignore it

File: src/model/model.cpp (hunk 5 above)
Comment (anarchuser) [reply src/model/model.cpp]: that's resolved

Comment (anarchuser) [reply src/model/model.cpp]: that's resolved

**Review by xeruf — APPROVED** [review]
